src/traits.rs

```diff
@@ -99,20 +110,84 @@ impl Lerp for u32 {
     type Ratio = f64;
 }
 
+impl Lerp for f32 {
+    type Ratio = f32;
+
+    fn lerp(a: Self, b: Self, ratio: Self::Ratio) -> Self {
+        a + (b - a) * ratio
+    }
+}
+
+
+pub trait Sample: Primitive + 'static {
+    const MAX_SAMPLE_VALUE: Self;
```

Comment:
Together with the above this is a bit odd. If it is an inherent property of `Component` then it must be the same for all usages in `Pixel`—including in alpha and non-alpha channels. This may be okay, or not? I don't have strong evidence either way.

Reply:
Good point. 

There's a few good places to put this value, but they are all more complex than this constant. I would define `0..1` as the default float range for all channels, and would only make it more complex once someone actually has a use case for it. Keeping it simple for now. 

Rename the constant to `DEFAULT_MAX_COMPONENT_VALUE` or even `DEFAULT_MAX_COMPONENT_VALUE_FOR_ALL_CHANNELS` would keep the door open for adding some kind of override later on (maybe in a color space definition, or even in the `invert(max_rgb: 1.0, max_alpha: 1.0)` function call).

Keep in mind that merging this pr will not set this decision in stone. It could still be added to the roadmap for the next release.